lib/Doctrine/ORM/Mapping/EscapingQuoteStrategy.php

```diff
@@ -0,0 +1,176 @@
+    {
+        if(isset($definition['quoted'])){
+            return $platform->quoteIdentifier($class->table['name']);
+        }
+        $reservedKeyList = $platform->getReservedKeywordsList();
+        if($reservedKeyList->isKeyword($definition['sequenceName'])){
+            return $platform->quoteIdentifier($definition['sequenceName']);
+        }
+        return $definition['sequenceName'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getJoinColumnName(array $joinColumn, ClassMetadata $class, AbstractPlatform $platform)
+    {
+        if(isset($joinColumn['quoted'])){
+            return $platform->quoteIdentifier($joinColumn['name']);
+        }
+        $reservedKeyList = $platform->getReservedKeywordsList();
+        if($reservedKeyList->isKeyword($joinColumn['name'])){
+            return $platform->quoteIdentifier($joinColumn['name']);
+        }
+        return $joinColumn['name'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getReferencedJoinColumnName(array $joinColumn, ClassMetadata $class, AbstractPlatform $platform)
+    {
+        if(isset($joinColumn['quoted'])){
+            return $platform->quoteIdentifier($joinColumn['referencedColumnName']);
+        }
+        $reservedKeyList = $platform->getReservedKeywordsList();
+        if($reservedKeyList->isKeyword($joinColumn['referencedColumnName'])){
+            return $platform->quoteIdentifier($joinColumn['referencedColumnName']);
+        }
+        return $joinColumn['referencedColumnName'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getJoinTableName(array $association, ClassMetadata $class, AbstractPlatform $platform)
+    {
+        if(isset($association['joinTable']['quoted'])){
+            return $platform->quoteIdentifier($association['joinTable']['name']);
+        }
+        $reservedKeyList = $platform->getReservedKeywordsList();
+        if($reservedKeyList->isKeyword($association['joinTable']['name'])){
+            return $platform->quoteIdentifier($association['joinTable']['name']);
+        }
+        return $association['joinTable']['name'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIdentifierColumnNames(ClassMetadata $class, AbstractPlatform $platform)
+    {
+        $quotedColumnNames = array();
+
+        foreach ($class->identifier as $fieldName) {
+            if (isset($class->fieldMappings[$fieldName])) {
+                $quotedColumnNames[] = $this->getColumnName($fieldName, $class, $platform);
+
+                continue;
+            }
+
+            // Association defined as Id field
+            $joinColumns            = $class->associationMappings[$fieldName]['joinColumns'];
+            $assocQuotedColumnNames = array_map(
+                function ($joinColumn) use ($platform)
+                {
+                    if(isset($joinColumn['quoted'])){
+                        return $platform->quoteIdentifier($joinColumn['name']);
```

Comment:
The variable $joinColumn['name'] don't seem to be quoted (if needed) to me, but I might have made a mistake.